packages/frontend-v2/pages/login.tsx

```diff
@@ -61,92 +38,65 @@ const LoginForm = () => {
       router.push("/home");
     } catch (err) {
       const error = err as AxiosError;
-      if (error.response?.status === 401) toast.error("Invalid Credentials!");
-      else toast.error("Something went wrong!");
+      if (error.response?.status === 401) {
+        toast.error("Invalid Credentials!");
+      } else {
+        handleApiClientError(error, router);
+      }
     }
   };
 
   if (renderSpinner) return <LoadingPage />;
 
   return (
-    <FormFormat onSubmit={handleSubmit(onSubmitHandler)}>
-      <LoginFormTopInput errors={errors} register={register} />
-      <LoginFormButton router={router} isSubmitting={isSubmitting} />
-    </FormFormat>
-  );
-};
-
-const LoginFormTopInput: React.FC<LoginFormTopProps> = ({
-  errors,
-  register,
-}) => {
-  return (
-    <HeaderAndInput>
-      <Text
-        fontSize="3xl"
-        color="primary.red.main"
-        as="b"
-        mb="xl"
-        textAlign="center"
-      >
-        Hey there!
-      </Text>
+    <AuthForm
+      onSubmit={handleSubmit(onSubmitHandler)}
+      headingText="Welcome Back!"
+      inputs={
```

Comment:
Instead of passing in React Nodes as props, would it be better to pass in the inputs as children?

Reply:
a solid case could be made for both! passing in as inputs makes it more meaningful here since this component isn't some layout or scaffolding but rather meant to render the entire form given the input fields. this wouldn't be clear if we rendered the children as the inputs of the form.